go-controller/pkg/clustermanager/node/node_allocator.go

```diff
@@ -411,10 +428,28 @@ func (na *NodeAllocator) Sync(nodes []interface{}) error {
 					klog.Errorf("Failed to parse hybrid overlay for node %s: %v", node.Name, err)
 				} else if hostSubnet != nil {
 					klog.V(5).Infof("Node %s contains subnets: %v", node.Name, hostSubnet)
-					if err := na.hybridOverlaySubnetAllocator.ReleaseNetworks(node.Name, hostSubnet); err != nil {
+					if err := na.hybridOverlaySubnetAllocator.MarkAllocatedNetworks(node.Name, hostSubnet); err != nil {
 						klog.Errorf("Failed to mark the subnet %v as allocated in the hybrid subnet allocator for node %s: %v", hostSubnet, node.Name, err)
 					}
 				}
+			} else if config.HybridOverlay.Enabled && config.HybridOverlay.ClusterSubnets == nil {
```

Comment:
looking at the implementation of `hasHybridOverlayAllocation()` have you missed the secondary network check here?

Reply:
right.